src/smc-webapp/frame-editors/html-editor/iframe-html.tsx

```diff
@@ -98,13 +101,24 @@ export class IFrameHTML extends Component<PropTypes, {}> {
   }
 
   render_iframe() {
+    if (
+      this.props.mode == "rmd" &&
+      this.props.derived_file_types != undefined
+    ) {
+      if (!this.props.derived_file_types.contains("html")) {
+        return this.render_no_html();
```

Comment:
just in "rmd" mode, and when we do know the situation about derived file types, and we've learned that the "html" version doesn't exist, this iframe won't be able to show the html file → then give the user a hint what is going on